src/core/backend.cc

```diff
@@ -135,6 +135,8 @@ InferenceBackend::SetModelConfig(
   metric_reporter_ = std::make_shared<MetricModelReporter>(
       Name(), version_, config_.metric_tags());
 
+  stats_collector_ = std::make_shared<StatsAggregator>(metric_reporter_);
```

Comment:
This doesn't need to be shared_ptr does it? It can just be a member of the InferenceBackend class (or a unique_ptr until metric_reporter_ gets sorted out).